ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-security.j2

```diff
@@ -63,6 +63,33 @@ management security
 {%             endif %}
 {%         endfor %}
 {%     endif %}
+{%     for profile in management_security.shared_secret_profiles | arista.avd.natural_sort("profile") %}
+   session shared-secret profile {{ profile.profile }}
+{%         for secret in profile.secrets | arista.avd.natural_sort("name") %}
+{%             if secret.secret is arista.avd.defined and
+                  ((secret.receive_lifetime.infinite is arista.avd.defined or (secret.receive_lifetime.start_date_time and secret.receive_lifetime.end_date_time is arista.avd.defined)) and
+                   (secret.transmit_lifetime.infinite is arista.avd.defined or (secret.transmit_lifetime.start_date_time and secret.transmit_lifetime.end_date_time is arista.avd.defined))) %}
+{%                 set secret_cli = "secret " ~ secret.name ~ " " ~ secret.secret_type | arista.avd.default("7") ~ " " ~ secret.secret | arista.avd.hide_passwords(hide_passwords) %}
+{%                 if secret.receive_lifetime.infinite is arista.avd.defined(true) %}
+{%                     set secret_cli = secret_cli ~ " receive-lifetime infinite" %}
+{%                 else %}
+{%                     set secret_cli = secret_cli ~ " receive-lifetime " ~ secret.receive_lifetime.start_date_time ~ " " ~ secret.receive_lifetime.end_date_time %}
+{%                 endif %}
+{%                 if secret.transmit_lifetime.infinite is arista.avd.defined(true) %}
+{%                     set secret_cli = secret_cli ~ " transmit-lifetime infinite" %}
+{%                 else %}
+{%                     set secret_cli = secret_cli ~ " transmit-lifetime " ~ secret.transmit_lifetime.start_date_time ~ " " ~ secret.transmit_lifetime.end_date_time %}
+{%                 endif %}
+{%                 if secret.local_time is arista.avd.defined(true) %}
+{%                     set secret_cli = secret_cli ~ " local-time" %}
+{%                 endif %}
+      {{ secret_cli }}
+{%             endif %}
+{%         endfor %}
+{%         if not loop.last %}
+   !
+{%         endif %}
```

Comment:
Please move this to insert before each profile. It should not be required to check the loop then.

Reply:
Fixed.